packages/traveling-fastlane/actions/manage_ad_hoc_provisioning_profile.rb

```diff
@@ -43,7 +43,31 @@ def register_missing_devices(udids)
 end
 
 def find_profile_by_bundle_id(bundle_id)
-  Spaceship::Portal.provisioning_profile.ad_hoc.find_by_bundle_id(bundle_id: bundle_id).first
+  expo_profiles = Spaceship::Portal.provisioning_profile.ad_hoc.find_by_bundle_id(bundle_id: bundle_id)
+  expo_profiles = expo_profiles.select{ |profile|
+    profile.name.start_with?('*[expo]') and profile.expires.to_datetime > DateTime.now
+  }
+  # find profiles associated with our development cert
+  expo_profiles_with_cert = expo_profiles.select{ |profile|
+    profile.certificates.any?{|cert| cert.raw_data['serialNumber'] == $certSerialNumber} # this api only returns field 'serialNumber', but not 'serialNum'
+  }
+
+  if !expo_profiles_with_cert.empty?
+    # there is an expo managed profile with our desired certificate
+    # return the profile that will be valid for the longest duration
+    expo_profiles_with_cert.sort_by{|profile| profile.expires}.last  
+  elsif !expo_profiles.empty? 
+    # there is an expo managed profile, but it doesnt have our desired certificate
+    # append the certificate and update the profile 
+    dist_cert = find_dist_cert($certSerialNumber, in_house?($teamId))
+    profile = expo_profiles.sort_by{|profile| profile.expires}.last
+    profile.certificates.push(dist_cert)
```

Comment:
Hmmmm, as far as I know, a provisioning profile can be associated with only one certificate. Please correct me if I'm wrong.

Reply:
hmm yeah so i was confused by this one too. The fastlane API returns an array of certs (`profile.certificates`), but in the Apple API, you can only associate one cert to the profile. In practice, the code above just replaces the old cert with the new cert. 

maybe it would be better to just clobber this field with `profile.certificates=[dist_cert]`?

Reply:
Yes, I'd prefer `profile.certificates = [dist_cert]` here.